ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/MigrationsUtil.java

```diff
@@ -47,6 +88,29 @@ public static Client getKsqlClient(final String ksqlServerUrl) throws MigrationE
         .setHost(url.getHost())
         .setPort(url.getPort());
 
-    return Client.create(options);
+    if (username != null || password != null) {
+      options.setBasicAuthCredentials(username, password);
+    }
+
+    // CHECKSTYLE_RULES.OFF: BooleanExpressionComplexity
+    final boolean useTls = !Strings.isNullOrEmpty(sslTrustStoreLocation)
```

Comment:
The logic here is not correct. It's possible that a user will want to use TLS even without any of these optional properties set. We should infer whether to use TLS or not based on the ksql server URL (whether it starts with HTTPS) instead.